common/db.py

```diff
@@ -10,3 +11,15 @@ class SubQuerySum(Subquery):
     def __init__(self, queryset, output_field=None, *, sum_field="", **extra):
         extra["sum_field"] = sum_field
         super().__init__(queryset, output_field, **extra)
+
+
+def raw_prefixed_query(text: str) -> SearchQuery:
+    """
+    Create a query that searches for each word separately and matched partial words if match is a prefix.
+    Remove any whitespace and replace any single quote mark with two quote marks.
+    https://www.postgresql.org/docs/current/datatype-textsearch.html#DATATYPE-TSQUERY
+    """
+    return SearchQuery(
+        " | ".join(f"'{val}':*" for value in text.split(" ") if (val := value.strip().replace("'", "''")) != ""),
+        search_type="raw",
+    )
```

Comment:
SonarCloud doesn't like this and we can get rid of the walrus operator and make it a bit cleaner like this:
```
    return SearchQuery(
        " | ".join(f"'{val.replace("'", "''")}':*" for value in text.split(" ") if value != ""),
        search_type="raw",
    )
```

- Since you are splitting the text by ` ` (space), there is no need to `strip` it since there will never be strings with the space character in it.
- The `.replace("'", "''")` can be moved left, since its not relevant in the equals check
- Finally what we have left is `... if value != ""`. Since empty strings evaluate to false, the `!= ""` can also be removed or kept for clarity's sake

Reply:
Good simplification, although I do unfortunately need to use tripe quotes for the f-string since I'm already using single and double ones:

```python
def raw_prefixed_query(text: str) -> SearchQuery:
    return SearchQuery(
        " | ".join(f"""'{value.replace("'", "''")}':*""" for value in text.split(" ") if value != ""),
        search_type="raw",
    )
```